webhooks-extension/pkg/endpoints/webhook.go

```diff
@@ -526,40 +520,54 @@ func (r Resource) createWebhook(request *restful.Request, response *restful.Resp
 	response.WriteHeader(http.StatusCreated)
 }
 
-func (r Resource) createIngressTaskRun(mode, installNS string) (*pipelinesv1alpha1.TaskRun, error) {
-	// Unlike webhook creation, the ingress does not need a protocol specified
-	callback := strings.TrimPrefix(r.Defaults.CallbackURL, "http://")
-	callback = strings.TrimPrefix(callback, "https://")
+func (r Resource) createDeleteIngress(mode, installNS string) error {
+	if mode == "create" {
+		// Unlike webhook creation, the ingress does not need a protocol specified
+		callback := strings.TrimPrefix(r.Defaults.CallbackURL, "http://")
+		callback = strings.TrimPrefix(callback, "https://")
 
-	params := []pipelinesv1alpha1.Param{
-		{Name: "Mode", Value: pipelinesv1alpha1.ArrayOrString{Type: pipelinesv1alpha1.ParamTypeString, StringVal: mode}},
-		{Name: "CallbackURL", Value: pipelinesv1alpha1.ArrayOrString{Type: pipelinesv1alpha1.ParamTypeString, StringVal: callback}},
-		{Name: "EventListenerServiceName", Value: pipelinesv1alpha1.ArrayOrString{Type: pipelinesv1alpha1.ParamTypeString, StringVal: "el-" + eventListenerName}},
-		{Name: "EventListenerPort", Value: pipelinesv1alpha1.ArrayOrString{Type: pipelinesv1alpha1.ParamTypeString, StringVal: "8080"}}}
-
-	ingressTaskRun := pipelinesv1alpha1.TaskRun{
-		ObjectMeta: metav1.ObjectMeta{
-			GenerateName: mode + "-ingress-",
-			Namespace:    installNS,
-		},
-		Spec: pipelinesv1alpha1.TaskRunSpec{
-			Inputs: pipelinesv1alpha1.TaskRunInputs{
-				Params: params,
+		ingress := &v1beta1.Ingress{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "el-" + eventListenerName,
+				Namespace: installNS,
 			},
-			ServiceAccount: os.Getenv("SERVICE_ACCOUNT"),
-			TaskRef: &pipelinesv1alpha1.TaskRef{
-				Name: "ingress-task",
+			Spec: v1beta1.IngressSpec{
+				Rules: []v1beta1.IngressRule{
+					{
+						Host: callback,
+						IngressRuleValue: v1beta1.IngressRuleValue{
+							HTTP: &v1beta1.HTTPIngressRuleValue{
+								Paths: []v1beta1.HTTPIngressPath{
+									{
+										Backend: v1beta1.IngressBackend{
+											ServiceName: "el-" + eventListenerName, 
+											ServicePort: intstr.IntOrString{
+												Type: intstr.Int,
+												IntVal: 8080,
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
 			},
-		},
-	}
-
-	tr, err := r.TektonClient.TektonV1alpha1().TaskRuns(installNS).Create(&ingressTaskRun)
-	if err != nil {
-		return &pipelinesv1alpha1.TaskRun{}, err
+		}
+		ingress, err := r.K8sClient.ExtensionsV1beta1().Ingresses(installNS).Create(ingress)
+		if err != nil {
+			return err
+		}
+		logging.Log.Debug("Ingress being created")
```

Comment:
At this point it is "Ingress has been created"

Reply:
I change it.